lineage/states/StateDistributionGamma.py

```diff
@@ -190,3 +189,35 @@ def time_censor(cell, desired_experiment_time):
             # the daughters are no longer observed
             cell.left.observed = False
             cell.right.observed = False
+
+
+def atonce_estimator(x_list, gammas_list):
+    """ User-defined way of estimating the parameters given a list of the tuples of observations from a group of cells.
+        gammas_list is only for one state. """
+    # unzipping the list of tuples
+    x_data = [np.array(x) for x in x_list]
+
+    # getting the observations as individual lists
+    bern_obs = [x[:, 0].astype('bool') for x in x_data]
+    γ_obs = [x[:, 1] for x in x_data]
+    gamma_obs_censor = [x[:, 2] for x in x_data]
+
+    b_masks = [np.isfinite(bern_) for bern_ in bern_obs]
+    bern_params = np.zeros(4)
+    for i, b_mask in enumerate(b_masks):
+        # Handle an empty state
+        if np.sum(gammas_list[i][b_mask]) == 0.0:
+            bern_params[i] = np.average(bern_obs[i][b_mask])
+        else:
+            bern_params[i] = np.average(bern_obs[i][b_mask], weights=gammas_list[i][b_mask])
+
+    # Both unoberved and dead cells should be removed from gamma
+    g_masks = [np.logical_and(np.isfinite(γ_o), berns) for γ_o, berns in zip(γ_obs, bern_obs)]
+    for g_mask in g_masks:
+        assert np.sum(g_mask) > 0, f"All the cells are eliminated from the Gamma estimator."
+
+    γ_obs_total = [g_obs[g_masks[i]] for i, g_obs in enumerate(γ_obs)]
+    γ_obs_total_censored = [g_obs_cen[g_masks[i]] for i, g_obs_cen in enumerate(gamma_obs_censor)]
+    gammas_total = [np.vstack(gamma_tot)[g_masks[i]] for i, gamma_tot in enumerate(gammas_list)]
+    output = gamma_estimator_atonce(γ_obs_total, γ_obs_total_censored, gammas_total)
```

Comment:
@Farnazmdi why would running the function twice help to speed it up?

Reply:
I know sorry, I am working on it in local, it should have been recursive, not running that twice. But the problem is that when I call the function inside it, it doesn't get out of it and just keeps running itself. I should find a good stopping point.

Reply:
Why would making it recursive help in any way?

Reply:
When I said "last iteration", I meant last iteration of Baum-Welch, not of the optimization here. Think through the problem. There is also the separated estimator as an example.

Reply:
It should be fixed now. I appreciate if you could review it.